src/mongocxx/test/spec/unified_tests/operations.hh

```diff
@@ -33,5 +33,13 @@ bsoncxx::document::value run(entity::map& map,
 
 }  // namespace operations
 
+namespace operations {
+
+bsoncxx::stdx::optional<read_concern> lookup_read_concern(bsoncxx::document::view doc);
+bsoncxx::stdx::optional<write_concern> lookup_write_concern(bsoncxx::document::view doc);
+bsoncxx::stdx::optional<read_preference> lookup_read_preference(bsoncxx::document::view doc);
```

Comment:
Are these functions used outside of `src/mongocxx/test/spec/unified_tests/operations.cpp`? If not, I suggest leaving them private to `operations.cpp`.

Reply:
There are actually multiple versions of them! These were the ones that do what's needed, so I've exposed them (it isn't really making the situation worse), and I was sure to mention them in CXX-2385:
https://jira.mongodb.org/browse/CXX-2385

Reply:
Ack. Thank you for noting that in CXX-2385.